File: CS/WindowsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Windows.Forms;

namespace WindowsApplication1
{
	public partial class Form1 : Form
	{
		public List<int> DisabledRowHandles;

		public Form1() {
			InitializeComponent();
			DisabledRowHandles = new List<int>() { 0, 3, 4, 7, 9 };
			gridControl1.DataSource = CreateTable(20);
			disabledCellEvents1.ProcessingCell += DisabledCellEvents1_ProcessingCell;
		}

		private static DataTable CreateTable(int RowCount) {
			DataTable tbl = new DataTable();
			tbl.Columns.Add("Name", typeof(string));
			tbl.Columns.Add("ID", typeof(int));
			tbl.Columns.Add("Number", typeof(int));
			tbl.Columns.Add("Date", typeof(DateTime));
			for(int i = 0; i < RowCount; i++)
				tbl.Rows.Add(new object[] { String.Format("Name{0}", i), i, 3 - i, DateTime.Now.AddDays(i) });
			return tbl;
		}
		private void DisabledCellEvents1_ProcessingCell(object sender, DevExpress.Utils.Behaviors.Common.ProcessCellEventArgs e) {
			if(DisabledRowHandles.Contains(e.RecordId))
				e.Disabled = true;
		}
	}
}

File: VB/WindowsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Windows.Forms

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form
		Public DisabledRowHandles As List(Of Integer)

		Public Sub New()
			InitializeComponent()
			DisabledRowHandles = New List(Of Integer) (New Integer() {0, 3, 4, 7, 9})
			gridControl1.DataSource = CreateTable(20)
			AddHandler disabledCellEvents1.ProcessingCell, AddressOf DisabledCellEvents1_ProcessingCell
		End Sub

		Private Shared Function CreateTable(ByVal RowCount As Integer) As DataTable
			Dim tbl As New DataTable()
			tbl.Columns.Add("Name", GetType(String))
			tbl.Columns.Add("ID", GetType(Integer))
			tbl.Columns.Add("Number", GetType(Integer))
			tbl.Columns.Add("Date", GetType(DateTime))
			For i As Integer = 0 To RowCount - 1
				tbl.Rows.Add(New Object() { String.Format("Name{0}", i), i, 3 - i, DateTime.Now.AddDays(i) })
			Next i
			Return tbl
		End Function
		Private Sub DisabledCellEvents1_ProcessingCell(ByVal sender As Object, ByVal e As DevExpress.Utils.Behaviors.Common.ProcessCellEventArgs)
			If DisabledRowHandles.Contains(e.RecordId) Then
				e.Disabled = True
			End If
		End Sub
	End Class
End Namespace